src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+const wallElement = document.querySelector('.wall');
+const spiderElement = document.querySelector('.spider');
+
+const centerSpider = () => {
+  const wallRect = wallElement.getBoundingClientRect();
+  const spiderRect = spiderElement.getBoundingClientRect();
+
+  const wallStyles = window.getComputedStyle(wallElement);
+
+  const wallBorderWidth =
+    parseFloat(wallStyles.borderLeftWidth) +
+    parseFloat(wallStyles.borderRightWidth);
+  const wallBorderHeight =
+    parseFloat(wallStyles.borderTopWidth) +
+    parseFloat(wallStyles.borderBottomWidth);
+
+  const wallHeight = wallRect.height;
+  const wallWidth = wallRect.width;
+
+  const spiderHeight = spiderRect.height;
+  const spiderWidth = spiderRect.width;
+
+  const spiderTop = (wallHeight - wallBorderHeight) / 2 - spiderHeight / 2;
+  const spiderLeft = (wallWidth - wallBorderWidth) / 2 - spiderWidth / 2;
+
+  spiderElement.style.top = `${spiderTop}px`;
+  spiderElement.style.left = `${spiderLeft}px`;
+};
+
+centerSpider();
```

Comment:
You can simplify like this 
```suggestion
const spiderContainer = document.querySelector('.wall');
const spider = document.querySelector('.spider');

const leftPosition = (spiderContainer.clientHeight - spider.clientWidth) / 2;
const topPosition = (spiderContainer.clientHeight - spider.clientHeight) / 2;

spider.style.left = `${leftPosition}px`;
spider.style.top = `${topPosition}px`;
```

Reply:
ty